examples/chainrules.jl

```diff
@@ -0,0 +1,156 @@
+using JuMP
+using Clp
+using DiffOpt
+using Test
+using ChainRulesCore
+
+# This script creates the JuMP model for a small unit commitment instance
+# represented in a solution map function taking parameters as arguments and returning
+# the optimal solution as output.
+
+# The derivatives of this solution map can then be expressed in ChainRules semantics
+# and implemented using DiffOpt
+
+ATOL=1e-4
+RTOL=1e-4
+
+"""
+Solution map of the problem using parameters:
+- `load1_demand, load2_demand` for the demand of two nodes
+- `gen_costs` is the vector of generator costs
+- `noload_costs` is the vector of fixed activation costs of the generators,
+and returning the optimal output power `p`.
+"""
+function unit_commitment(load1_demand, load2_demand, gen_costs, noload_costs; model = Model(() -> diff_optimizer(Clp.Optimizer)))
```

Comment:
I think we can just do
```suggestion
function unit_commitment(load1_demand, load2_demand, gen_costs, noload_costs; model = Clp.Optimizer)
```